docs/src/main/tut/datatypes/validated.md

```diff
@@ -256,47 +259,97 @@ For example:
 
 ```tut:book
 FormValidatorNel.validateForm(
-            username = "Joe",
-            password = "Passw0r$1234",
-            firstName = "John",
-            lastName = "Doe",
-            age = 21
-          )
+  username = "Joe",
+  password = "Passw0r$1234",
+  firstName = "John",
+  lastName = "Doe",
+  age = 21
+)
 
 FormValidatorNel.validateForm(
-            username = "Joe%%%",
-            password = "password",
-            firstName = "John",
-            lastName = "Doe",
-            age = 21
-          )
+  username = "Joe%%%",
+  password = "password",
+  firstName = "John",
+  lastName = "Doe",
+  age = 21
+)
 ```
 
 Sweet success! Now you can take your validation process to the next level!
 
-### Coming from `Either`-based validation
+### A short detour
+
+Typically, you'll see that `Validated` will be accompanied by a `NonEmptyList` when it comes to accumulation. The thing here is that you can define your own accumulative data structure and you're not limited to the aforementioned construction.
+
+For doing this, you have to provide a `Semigroup` instance. `NonEmptyList`, by definition has its own `Semigroup`. For those who don't know what a `Semigroup` is, let's see a simple example.
```

Comment:
This is really great, thank you! However, you didn't really include an example?

Reply:
It was more about how `ap` works for `Invalid`, but I can also provide a simple example of `Semigroup`. What do you think about giving the example first and then point to the actual behavior, as it is written now?